tor_archivist/main.py

```diff
@@ -54,86 +39,61 @@ def noop(cfg):
         # CTRL+C quickly: trigger an event loop every minute during wait
         # time.
         time.sleep(5)
         return
 
     logging.info('Starting archiving of old posts...')
-    # TODO the bot will now check ALL posts on the subreddit.
-    # when we remove old transcription requests, there aren't too many left.
-    # but we should make it stop after a certain amount of time anyway
-    # eg. if it encounters a post >36 hours old, it will break the loop
-
-    # TODO we can use .submissions(end=unixtime) apparently
-    for post in cfg.tor.new(limit=1000):
-
-        # [META] - do nothing
-        # [UNCLAIMED] - remove
-        # [COMPLETED] - remove and x-post to r/tor_archive
-        # [IN PROGRESS] - do nothing (should discuss)
-        # [DISREGARD] - remove
-        flair = post.link_flair_css_class
-
-        # is it a disregard post? Nuke it and move on -- we don't want those
-        # sitting around and cluttering up the sub
-        if flair == css_flair.disregard:
-            logging.info(f'Post "{post.title}" is marked as "Disregard", removing.')
-            post.mod.remove()
-            continue
-
-        if flair not in (css_flair.unclaimed, css_flair.completed):
-            logging.info(f'Post "{post.title}" is not completed, skipping removal.')
-            continue
-
-        # the original post that might have been transcribed
-        original_post = config.r.submission(url=post.url)
-
-        # find the original post subreddit
-        post_subreddit = subreddit_from_url(post.url)
 
-        # hours until a post from this subreddit should be archived
-        hours = cfg.archive_time_subreddits.get(
-            post_subreddit, cfg.archive_time_default)
-
-        # time since this post was made
-        date = datetime.utcfromtimestamp(post.created_utc)
-        seconds = int((datetime.utcnow() - date).total_seconds())
-
-        if CLEAR_THE_QUEUE_MODE:
-            logging.info(f'Removing "{post.title}" because Clear The Queue')
-            post.mod.remove()
-
-        elif seconds > hours * 3600:
-            logging.info(f'Post "{post.title}" is older than maximum age of {hours} hours, removing.')
-
-            post.mod.remove()
-
-        elif is_removed(original_post, full_check=True):
-            logging.info(f'Post "{original_post.title}" looks like it was removed on the other side. Nuking.')
-            post.mod.remove()
 
+    if unarchived_submissions := b_api.get("/submission/unarchived/").json():
+        if unarchived_submissions.get('data'):
+            for submission in unarchived_submissions['data']:
+
+                post_subreddit = subreddit_from_url(submission['url'])
+
+                # the original post from r/ToR that might have been transcribed
+                reddit_post = config.r.submission(id=submission['submission_id'])
+
+                # unccomment the below line in prodution to remove original r/ToR post
+                # reddit_post.mod.remove()
```

Comment:
That's gonna have to change. We don't live-edit source code before running in production. Or at least we try not to. 😅

Reply:
Fair enough, I wasn't really sure how to make it clear that we would need to add code in production to remove the original post.

Reply:
Let's hide this behind the DEBUG flag.